File: Assets/zfoolua/udp/UdpHelloResponse.lua

local UdpHelloResponse = {}

function UdpHelloResponse:new()
    local obj = {
        message = "" -- string
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function UdpHelloResponse:protocolId()
    return 1201
end

function UdpHelloResponse:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
    buffer:writeString(packet.message)
end

function UdpHelloResponse:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = UdpHelloResponse:new()
    local result0 = buffer:readString()
    packet.message = result0
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return UdpHelloResponse

File: Assets/zfoocs/Udp/UdpHelloResponse.cs
using System;
using System.Collections.Generic;

namespace zfoocs
{
    
    public class UdpHelloResponse
    {
        public string message;

        public static UdpHelloResponse ValueOf(string message)
        {
            var packet = new UdpHelloResponse();
            packet.message = message;
            return packet;
        }
    }


    public class UdpHelloResponseRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 1201;
        }

        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            UdpHelloResponse message = (UdpHelloResponse) packet;
            buffer.WriteInt(-1);
            buffer.WriteString(message.message);
        }

        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.ReadOffset();
            UdpHelloResponse packet = new UdpHelloResponse();
            string result0 = buffer.ReadString();
            packet.message = result0;
            if (length > 0) {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}
